gradle-plugins/src/main/kotlin/com/arcgismaps/CopySampleFilesTask.kt

```diff
@@ -42,6 +48,58 @@ class CopySampleFilesTask : Plugin<Project> {
 
             // Place in the assets codeFiles folder of the app.
             into(outputDir)
+
+            // After the copy finishes, run JSON generation + cleanup
+            doLast {
```

Comment:
I'm not great with gradle--what's the advantage of doing this part of the work at task execution time instead of configuration time

Reply:
Placing the code for the JSON generation and cleanup in a `doLast` block ensures that it only happens if/when the `copyCodeFiles` configuration task actually runs. This avoids doing cleanup on every single Gradle invocation.

If the task is up-to-date and Gradle skips it, the code in `doLast` won't run. This is helpful because we don’t want to regenerate the JSON or delete files every time Gradle configures tasks. Gradle can quickly skip tasks that are up-to-date during execution time.

Reply:
Awesome, sounds good. Thanks @shubham7109